server/api/db/sqldb/db.py

```diff
@@ -4721,6 +4724,48 @@ def delete_table_records_by_name(
             table_name=sanitized_table_name,
         )
 
+    def delete_index_by_name(
+        self,
+        session: Session,
+        index_name: str,
+        table_name: str,
+        raise_on_not_exists: bool = False,
+    ):
+        """
+        Delete an index by its name
+
+        :param session:             SQLAlchemy session
+        :param index_name:          Index name
+        :param table_name:          Table name
+        :param raise_on_not_exists: Raise an error if the index does not exist
+        """
+
+        # sanitize table and index names to prevent SQL injection, by removing all non-alphanumeric
+        # characters or underscores
+        sanitized_index_name = re.sub(r"[^a-zA-Z0-9_]", "", index_name)
+        sanitized_table_name = re.sub(r"[^a-zA-Z0-9_]", "", table_name)
+
+        # checking if the index exists can also help prevent SQL injection
+        if self._is_index_exists(
+            index_name=sanitized_index_name, table_name=sanitized_table_name
+        ):
+            drop_index_statement = text(
+                f"DROP INDEX {sanitized_index_name} ON {sanitized_table_name}"
+            )
+            session.execute(drop_index_statement)
```

Comment:
```suggestion
                from sqlalchemy import text, ...
                query = text("DROP INDEX :index_name ON :table_name")
                params = {
                    "index_name": "...",
                    "table_name": "...",
                }
                session.execute(query, params)
```
and now you dont need to sanitize anything as execute will perform sql-injection validation

Reply:
Turns out this is not possible because query params are used for values and not identifiers/names (they will get quoted)

Reply:
you can use `execute("DROP INDEX %s on %s", (index_name, table_name))` for selection iirc

Reply:
if it's decided that we delete manually then I'm removing this as well